presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -84,10 +93,42 @@ public PlanNodeStatsEstimate filterStats(
             Session session,
             TypeProvider types)
     {
-        return new FilterExpressionStatsCalculatingVisitor(statsEstimate, session, types).process(predicate)
+        Expression simplifiedExpression = simplifyExpression(session, predicate, types);
+        return new FilterExpressionStatsCalculatingVisitor(statsEstimate, session, types)
+                .process(simplifiedExpression)
                 .orElseGet(() -> normalizer.normalize(filterStatsForUnknownExpression(statsEstimate), types));
     }
 
+    private Expression simplifyExpression(Session session, Expression predicate, TypeProvider types)
+    {
+        Map<NodeRef<Expression>, Type> expressionTypes = getExpressionTypes(session, predicate, types);
+        ExpressionInterpreter interpreter = ExpressionInterpreter.expressionOptimizer(predicate, metadata, session, expressionTypes);
+        Object value = interpreter.optimize(NoOpSymbolResolver.INSTANCE);
+
+        if (value instanceof Expression) {
```

Comment:
`literalEncoder.toExpression()` already returns the Expression if the value is an instanceof Expression, and a NullLiteral if it equals null.  

I would get rid of the if statements and always return `literalEncoder.toExpression(value,BOOLEAN)`.  And add a `visitNullLiteral` to FilterExpressionStatsCalculatingVisitor.